File: Easy/Java/2293. Min Max Game.java
class Solution {
    public int minMaxGame(int[] nums) {
        if(nums.length == 1)
            return nums[0];
        
        int[] minMaxed = new int[nums.length / 2];
        for(int i = 0; i < minMaxed.length; i++){
            if(i % 2 == 0)
                minMaxed[i] = Math.min(nums[2 * i], nums[2 * i + 1]);
            else
                minMaxed[i] = Math.max(nums[2 * i], nums[2 * i + 1]);
        }

        return minMaxGame(minMaxed);
    }
}

File: Easy/Python/2293. Min Max Game.py
class Solution(object):
    def minMaxGame(self, nums):
        if len(nums) == 1:
            return nums[0]
        
        minMaxed = [0] * (len(nums) / 2)
        for i in range(len(minMaxed)):
            if i % 2 == 0:
                minMaxed[i] = min(nums[2 * i], nums[2 * i + 1])
            else:
                minMaxed[i] = max(nums[2 * i], nums[2 * i + 1])
        
        return self.minMaxGame(minMaxed)
